aptos-move/aptos-release-builder/src/validate.rs

```diff
@@ -0,0 +1,379 @@
+// Copyright (c) Aptos
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::ReleaseConfig;
+use anyhow::Result;
+use aptos_api_types::U64;
+use aptos_crypto::ed25519::Ed25519PrivateKey;
+use aptos_genesis::keys::PrivateIdentity;
+use aptos_rest_client::Client;
+use aptos_temppath::TempPath;
+use aptos_types::account_address::AccountAddress;
+use futures::executor::block_on;
+use serde::Deserialize;
+use std::{
+    fs,
+    path::{Path, PathBuf},
+    process::Stdio,
+    thread::sleep,
+    time::Duration,
+};
+use url::Url;
+
+#[derive(Clone, Debug)]
+pub struct NetworkConfig {
+    endpoint: Url,
+    root_key_path: PathBuf,
+    validator_account: AccountAddress,
+    validator_key: Ed25519PrivateKey,
+}
+
+#[derive(Deserialize)]
+struct CreateProposalEvent {
+    proposal_id: U64,
+}
+
+impl NetworkConfig {
+    pub fn new_from_dir(endpoint: Url, test_dir: PathBuf) -> Result<Self> {
+        let root_key_path = test_dir.join("mint.key");
+        let private_identity_file = test_dir.join("0/private-identity.yaml");
+        let private_identity =
+            serde_yaml::from_slice::<PrivateIdentity>(&fs::read(private_identity_file)?)?;
+
+        Ok(Self {
+            endpoint,
+            root_key_path,
+            validator_account: private_identity.account_address,
+            validator_key: private_identity.account_private_key,
+        })
+    }
+
+    pub fn submit_and_execute_proposal(&self, script_path: Vec<PathBuf>) -> Result<()> {
+        for path in script_path {
```

Comment:
For future, may be we can create the proposals in parallel with some buffer time in between. Waiting 30 secs after each proposal creation + vote makes things really really slow when there are multiple proposals.